modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/RenameProcessor.java

```diff
@@ -61,6 +61,22 @@ public void execute(IngestDocument document) {
 
         Object oldValue = document.getFieldValue(field, Object.class);
         document.setFieldValue(targetField, oldValue);
+        if (targetField.startsWith(field)) {
```

Comment:
is this really sufficient to describe a nested field relationship?

e.g. `targetField = "abcde"` and `field = "abc"`